app/services/vot/legacy_component_values.rb

```diff
@@ -0,0 +1,84 @@
+module Vot
+  module LegacyComponentValues
+    ## Identity proofing ACR values
+    LOA1 = ComponentValue.new(
+      name: Saml::Idp::Constants::LOA1_AUTHN_CONTEXT_CLASSREF,
+      description: 'Legacy LOA1',
+      implied_component_values: [],
+      requirements: [],
+    )
+    LOA3 = ComponentValue.new(
+      name: Saml::Idp::Constants::LOA3_AUTHN_CONTEXT_CLASSREF,
+      description: 'Legacy LOA3',
+      implied_component_values: [],
+      requirements: [:aal2, :identity_proofing],
+    )
+    IAL1 = ComponentValue.new(
+      name: Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF,
+      description: 'Legacy IAL1',
+      implied_component_values: [],
+      requirements: [],
+    )
+    IAL2 = ComponentValue.new(
+      name: Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF,
+      description: 'Legacy IAL2',
+      implied_component_values: [],
+      requirements: [:aal2, :identity_proofing],
+    )
+    IALMAX = ComponentValue.new(
+      name: Saml::Idp::Constants::IALMAX_AUTHN_CONTEXT_CLASSREF,
+      description: 'Legacy IALMAX',
+      implied_component_values: [],
+      requirements: [:aal2, :ialmax],
+    )
+
+    ## Authentication ACR values
+    DEFAULT = ComponentValue.new(
+      name: Saml::Idp::Constants::DEFAULT_AAL_AUTHN_CONTEXT_CLASSREF,
+      description: 'Legacy default authentication',
+      implied_component_values: [],
+      requirements: [],
+    )
+    AAL1 = ComponentValue.new(
+      name: Saml::Idp::Constants::AAL1_AUTHN_CONTEXT_CLASSREF,
+      description: 'Legacy AAL1',
+      implied_component_values: [],
+      requirements: [],
+    )
+    AAL2 = ComponentValue.new(
+      name: Saml::Idp::Constants::AAL2_AUTHN_CONTEXT_CLASSREF,
+      description: 'Legacy AAL2',
+      implied_component_values: [],
+      requirements: [:aal2],
+    )
+    AAL2_PHISHING_RESISTANT = ComponentValue.new(
+      name: Saml::Idp::Constants::AAL2_PHISHING_RESISTANT_AUTHN_CONTEXT_CLASSREF,
+      description: 'Legacy AAL2 with phishing resitance',
+      implied_component_values: [],
+      requirements: [:aal2, :phishing_resistant],
+    )
+    AAL2_HSPD12 = ComponentValue.new(
+      name: Saml::Idp::Constants::AAL2_HSPD12_AUTHN_CONTEXT_CLASSREF,
+      description: 'Legacy AAL2 with HSPD12',
+      implied_component_values: [],
+      requirements: [:aal2, :hspd12],
+    )
+    AAL3 = ComponentValue.new(
+      name: Saml::Idp::Constants::AAL3_AUTHN_CONTEXT_CLASSREF,
+      description: 'Legacy AAL3',
+      implied_component_values: [],
+      requirements: [:aal2, :hspd12],
+    )
+    AAL3_HSPD12 = ComponentValue.new(
+      name: Saml::Idp::Constants::AAL3_HSPD12_AUTHN_CONTEXT_CLASSREF,
+      description: 'Legacy AAL3 with HSPD12',
+      implied_component_values: [],
+      requirements: [:aal2, :hspd12],
+    )
+
+    MAPPED_BY_NAME = constants.map do |constant|
```

Comment:
Rather than `constants` magic, could we have a single constant as an array of values, also since we don't care about the actual constant name?

```rb
    VALUES = [
      ComponentValue.new(...)
      # ...
    ]

    MAPPED_BY_NAME = VALUES.map do |constant|
```

Reply:
LOL we just undid an approach like that because with named constants, they can reference each other instead of needing a separate lookup

Reply:
Ah, overlooked the "reference each other" aspect. Makes sense!